internal/cortex/querier/queryrange/queryrange.proto

```diff
@@ -56,6 +56,7 @@ message PrometheusInstantQueryResponse {
   string ErrorType = 3 [(gogoproto.jsontag) = "errorType,omitempty"];
   string Error = 4 [(gogoproto.jsontag) = "error,omitempty"];
   repeated PrometheusResponseHeader Headers = 5 [(gogoproto.jsontag) = "-"];
+  string Query = 6 [(gogoproto.jsontag) = "-"];
```

Comment:
Is this the original query that comes from the user? If yes, I wonder if we can somehow inject the variable instead of passing it through protobuf.

Reply:
I think this is possible if we change the `Merger` interface:
```
// Merger is used by middlewares making multiple requests to merge back all responses into a single one.
type Merger interface {
	// MergeResponse merges responses from multiple requests into a single Response
	MergeResponse(...Response) (Response, error)
}
```

Reply:
I am okay with changing this, not sure if this interface is used in Cortex though. Thoughts @yeya24?

Reply:
Makes sense to change the interface here.
Cortex is okay, Thanos and Cortex both needs this feature.